mitodl/mitxonline
# Hubspot integration

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1275 

#### What's this PR do?
Adds hubspot integration code to mitxonline (sync orders, products, users to mitxonline)

#### How should this be manually tested?
You'll need some new .env settings:
```
MITOL_HUBSPOT_API_PRIVATE_TOKEN=<ask_me, and I can also add you to the hubspot account>
MITOL_HUBSPOT_API_ID_PREFIX=<your_initials>-mitxonline-dev
HUBSPOT_PIPELINE_ID=19817792
```
Run the following, it should complete without errors:
```
docker-compose run web python manage.py sync_db_to_hubspot --products
```

Register a new user.  Log in to Hubspot and verify that contact with the same email exists there.

Update some details (like last name) and submit.  Check that the contact in hubspot is also updated.

Enroll in a course.  Wait a half minute or so then check that a deal corresponding to your order exists on hubspot.
